unit-tests/src/test/scala/java/lang/ThrowablesSuite.scala

```diff
@@ -0,0 +1,356 @@
+package java.lang
+
+// Portions of this Suite were ported, with thanks & gratitude,
+// from Scala.js testsuite/javalib/lang/ThrowablesTestOnJDK7.scala
+//
+// The rest is an original contribution to Scala Native.
+
+object ThrowablesSuite extends tests.Suite {
+
+  // Consolidate boilerplate; aids consistency.
+
+  private def getThrowableMessage(throwable: Throwable): String = {
+    if (throwable == null) "<null>" else throwable.getMessage
+  }
+
+  private def checkCause(throwable: Throwable,
+                         expectedCause: Throwable): Unit = {
+    val resultCause = throwable.getCause
+
+    val causeMessage = getThrowableMessage(throwable)
+
+    val expectedMessage = getThrowableMessage(expectedCause)
+
+    assert(resultCause == expectedCause,
+           s"cause: '${causeMessage}' != expected: '${expectedMessage}'")
+  }
+
+  private def checkMessage(throwable: Throwable,
+                           expectedMessage: String): Unit = {
+    val resultMessage = throwable.getMessage
+
+    assert(resultMessage == expectedMessage,
+           s"message: '${resultMessage}' != expected: '${expectedMessage}'")
+  }
+
+  private def checkStackTraceString(trace: String): Unit = {
+    val startText = "java.lang.Throwable"
+    assert(trace.startsWith(startText),
+           s"Expected trace to start with '${startText}' and it did not.")
+
+    val containsText = "\tat <none>.main(Unknown Source)"
+    assert(trace.contains(containsText),
+           s"Expected trace to contain '${containsText}' and it did not.")
+  }
+
+  private def checkStackTrace(throwable: Throwable): Unit = {
+    val sw = new java.io.StringWriter
+    val pw = new java.io.PrintWriter(sw)
+
+    throwable.printStackTrace(pw)
+
+    checkStackTraceString(sw.toString)
+  }
+
+  private def checkSuppressed(throwable: Throwable, expectedLength: Int) {
+    val getSuppressedLength = throwable.getSuppressed.length
+    assert(getSuppressedLength == expectedLength,
+           s"getSuppressed.length: ${getSuppressedLength} != " +
+             s"expected: ${expectedLength}")
+  }
+
+  private def checkConstructed(throwable: Throwable,
+                               expectedMessage: String,
+                               expectedCause: Throwable,
+                               expectedSuppressedLength: Int): Unit = {
+
+    checkMessage(throwable, expectedMessage)
+    checkCause(throwable, expectedCause)
+    checkSuppressed(throwable, 0)
+    checkStackTrace(throwable)
+  }
+
+  // Zero & two argument constructor tests will exercise cases where
+  // the rightmost two arguments are true.
+
+  test("Throwable(message, cause, false, false)") {
+    val expectedMessage = "Athchomar chomakea"
+    val expectedCause   = new Throwable("Khal Drogo")
+
+    val throwable = new Throwable(expectedMessage, expectedCause, false, false)
+
+    checkConstructed(throwable, expectedMessage, expectedCause, 0)
+
```

Comment:
These blank lines at the end of tests are unnecessary.

In general, avoid blank lines just after `{` and just before `}`, except in some cases of long `class`es and `object`s.

Reply:
Thank you for the simple & clear decision rule.  I have made a local copy
and will use that practice going forward.

I swept through the file and deleted blank lines following that rule.  I think I got them
all.